### PR TITLE
Fix Dawn4Py SIR serialization utils to work with GTPy

### PR DESCRIPTION
**Pull request by eddie-c-davis**
## Technical Description

This PR resolves the following GT4Py errors in the SIR serialization step in the Dawn4Py integration that occur in the FV3 stencils:

```python
  File "/home/eddied/Work/dawn/dawn/src/dawn4py/serialization/utils.py", line 357, in make_stmt
    raise SIRError("cannot create Stmt from type {}".format(type(stmt)))
dawn4py.serialization.error.SIRError: cannot create Stmt from type <class 'SIR.statements_pb2.Field'>
```

... and ...

```python
File "/home/eddied/Work/dawn/dawn/src/dawn4py/serialization/utils.py", line 722, in make_literal_access_expr
  builtin_type.type_id = type
  AttributeError: 'BuiltinType' object attribute 'type_id' is read-only
```

### CODE REVIEW

**Comment by eddie-c-davis** [comment]
launch jenkins

File: dawn/src/dawn4py/serialization/utils.py [hunk 1]
```diff
@@ -718,7 +718,8 @@ def make_literal_access_expr(value: str, type: BuiltinType.TypeID) -> LiteralAcc
     :param type:    Builtin type id of the literal.
     """
     builtin_type = BuiltinType()
-    builtin_type.type_id = type
+    if builtin_type.type_id != type:
```
Comment (jdahm) [inline dawn/src/dawn4py/serialization/utils.py]: Is this change necessary?

Comment (eddie-c-davis) [reply dawn/src/dawn4py/serialization/utils.py]: Yes, it throws the second error in the description otherwise.

Comment (jdahm) [reply dawn/src/dawn4py/serialization/utils.py]: Ah ok, an error gets raised in the = operation otherwise. Thanks for clarifying!
